src/lib.rs

```diff
@@ -1,6 +1,189 @@
+    ///
+    /// assert!(var.clone().isnt("BAZ").assign("BAZ").var("FOO") != var);
+    ///
+    /// ```
+    pub fn assign<T: ToString>(mut self, value: T) -> Environment {
+        if self.condition {
+            self.value = OsString::from(value.to_string());
+        }
+
+        self.environment.vars.push((self.key, self.value));
+
+        self.environment
+    }
+
+    /// Clear the value of a variable.
+    ///
+    /// Note: clearing can be conditioned by
+    /// [is](struct.Var.html#method.is) or [isnt](struct.Var.html#method.isnt)
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// extern crate environment;
+    ///
+    /// let var: environment::Var = ("FOO", "BAR").into();
+    ///
+    /// assert_eq!(var.clone().is("BAR").clear().var("FOO"), ("FOO", "").into());
+    ///
+    /// assert_eq!(var.clone().isnt("BAR").clear().var("FOO"), var);
+    ///
+    /// ```
+    pub fn clear(mut self) -> Environment {
+        if self.condition {
+            self.value.clear();
+        }
+
+        self.environment.vars.push((self.key, self.value));
+
+        self.environment
+    }
+
+    /// Append a value to the current variable's value.
+    ///
+    /// Note: appending can be conditioned by
+    /// [is](struct.Var.html#method.is) or [isnt](struct.Var.html#method.isnt)
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// extern crate environment;
+    ///
+    /// let var: environment::Var = ("FOO", "BAR").into();
+    ///
+    /// assert_eq!(var.clone().is("BAR").append("BAZ").var("FOO"), ("FOO", "BARBAZ").into());
+    ///
+    /// assert_eq!(var.clone().isnt("BAR").append("BAZ").var("FOO"), var);
+    ///
+    /// ```
+    pub fn append<T: ToString>(mut self, value: T) -> Environment {
+        if self.condition {
+            self.value.push(value.to_string());
+        }
+
+        self.environment.vars.push((self.key, self.value));
+
+        self.environment
+    }
+
+    /// Define the variable's value equal to the system variable.
+    ///
+    /// Note: inheriting can be conditioned by
+    /// [is](struct.Var.html#method.is) or [isnt](struct.Var.html#method.isnt)
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// extern crate environment;
+    ///
+    /// ::std::env::set_var("FOO", "BAR");
+    ///
+    /// let var: environment::Var = ("FOO", "BAZ").into();
+    ///
+    /// assert_eq!(var.clone().is("BAZ").inherit().var("FOO"), ("FOO", "BAR").into());
+    /// assert_eq!(var.clone().isnt("BAZ").inherit().var("FOO"), ("FOO", "BAZ").into());
+    ///
+    /// ```
+    pub fn inherit(mut self) -> Environment {
+        if self.condition {
+            self.value = if let Some(v) = ::std::env::vars_os()
+                .filter(|&(ref k, _)| k == &self.key)
+                .collect::<Vec<(OsString, OsString)>>()
+                .first()
+            {
+                v.1.clone()
+            } else {
+                OsString::new()
```

Comment:
If it assigns a blank string on failure to look up, that should probably be noted.

Alternatively, should it error?